### PR TITLE
Fix/login error message

### PR DESCRIPTION
**Pull request by thiessenp-cds**
# Summary | Résumé

Updates the login page to handle how errors are received. Only the logic was changed, not the error messages etc.

# Test instructions | Instructions pour tester la modification

Try an invalid login. An auth related error should show.
You could try programmatically throwing different errors from the API to see the right error message displayed in the UI. Maybe there is a better way though.

Try creating a new user. Probably the main thing to test is that this still works.

### CODE REVIEW

File: lib/hooks/auth/useLogin.ts [hunk 1]
```diff
@@ -37,24 +37,36 @@ export const useLogin = () => {
         timeout: process.env.NODE_ENV === "production" ? 60000 : 0,
       });
 
-      if (data?.error) {
-        const error = data.error;
-        if (hasError("UserNotConfirmedException", error)) {
+      // TODO: are there other challengeState/states to check for? It looks like the rest are errors (API responses)
```
Comment (thiessenp-cds) [inline lib/hooks/auth/useLogin.ts:40]: @craigzour from what I can see in [...nextauth].ts this is the only success message. Would you know if this is right?

Comment (craigzour) [reply lib/hooks/auth/useLogin.ts:40]: And yes the `challengeState` can only have the `MFA` value associated to it for now.

Comment (thiessenp-cds) [reply lib/hooks/auth/useLogin.ts:40]: Got it - thanks!

File: cypress.config.ts [hunk 2]
```diff
@@ -14,4 +14,12 @@ export default defineConfig({
       bundler: "webpack",
     },
   },
+  retries: {
```
Comment (thiessenp-cds) [inline cypress.config.ts:17]: @bryan-robitaille thanks this helps!

File: lib/hooks/auth/useLogin.ts [hunk 3]
```diff
@@ -37,24 +37,42 @@ export const useLogin = () => {
...
+    } catch (err: unknown) {
+      logMessage.error(err);
+
+      // 40X Client errors
+      if (axios.isAxiosError(err) && err.response && err.response.data) {
+        const { reason } = err.response.data;
+
+        // Handle slightly different response objects with the 400 check
+        if (
+          hasError(["UserNotFoundException", "NotAuthorizedException"], reason) ||
```
[the remark this answers is not in the document]
Comment (thiessenp-cds) [reply lib/hooks/auth/useLogin.ts:56]: I like it, definitely cleaner

**Review by timarney — APPROVED** [review]
From my testing works as advertised.

**Pull request merged**